views/js/initCardPayment.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+document.addEventListener('DOMContentLoaded', () => {
+  const interval = setInterval(() => {
+    if (window.paypalSdkPsCheckout !== undefined) {
+      initPsCheckout();
+      clearInterval(interval);
+    }
+  }, 200);
+});
+
+function initPsCheckout() {
+  if (typeof paypalOrderId === 'undefined') {
+    throw new Error('No paypal order id');
+  }
+
+  hostedFieldsErrors = JSON.parse(hostedFieldsErrors.replace(/&quot;/g, '"'));
+
+  initHostedFields();
+}
+
+function initHostedFields() {
+  // remove "amp;" from the url
+  const orderValidationLinkByCard = validateOrderLinkByCard.replace(/\amp;/g, '');
+
+  // check whether hosted fields is eligible for that Partner Account
+  if (paypalSdkPsCheckout.HostedFields.isEligible()) {
+    // render hosted fields
+    paypalSdkPsCheckout.HostedFields.render({
+      createOrder() {
+        return paypalOrderId;
+      },
+      styles: {
+        input: {
+          height: '25px',
+          'font-size': '1rem',
+        },
+        ':focus': {
+          'border-color': 'red',
+        },
+        'input.invalid': {
+          color: '#c05c67',
+        },
+      },
+      fields: {
+        number: {
+          selector: '#card-number',
+          placeholder: cardNumberPlaceholder,
+          class: 'form-control',
+        },
+        cvv: {
+          selector: '#cvv',
+          placeholder: cvvPlaceholder,
+        },
+        expirationDate: {
+          selector: '#expiration-date',
+          placeholder: expDatePlaceholder,
+        },
+      },
+    }).then((hf) => {
```

Comment:
hf ?

Reply:
hf = hosted fields = card payment :D It's obvious no ? ^^

Reply:
Changed :)